localstack/services/awslambda/resource_providers/aws_lambda_function.py

```diff
@@ -0,0 +1,510 @@
+# LocalStack Resource Provider Scaffolding v2
+from __future__ import annotations
+
+import os
+from pathlib import Path
+from typing import Optional, Type, TypedDict
+
+import localstack.services.cloudformation.provider_utils as util
+from localstack.services.cloudformation.resource_provider import (
+    CloudFormationResourceProviderPlugin,
+    OperationStatus,
+    ProgressEvent,
+    ResourceProvider,
+    ResourceRequest,
+)
+from localstack.services.lambda_.lambda_utils import get_handler_file_from_name
+from localstack.utils.archives import is_zip_file
+from localstack.utils.files import mkdir, new_tmp_dir, rm_rf, save_file
+from localstack.utils.strings import is_base64, to_bytes
+from localstack.utils.testutil import create_zip_file
```

Comment:
it's not very urgent but at some point these need to be inlined or included in `localstack.services.cloudformation.provider_utils`